src/dlls/mscoree/mscoree.cpp

```diff
@@ -792,7 +792,13 @@ BOOL PAL_GetPALDirectory(__out_ecount(cchBuffer) LPWSTR pbuffer,
 
     HRESULT hr = S_OK;
 
-    WCHAR pPath[MAX_LONGPATH];
+    WCHAR * pPath = new (nothrow) WCHAR[MAX_LONGPATH];
+    if (pPath == NULL)
+    {
+        WCHAR pPathStack[MAX_LONGPATH];
```

Comment:
I am not sure I understand the purpose of this construct. The pPathStack will be allocated always on the stack, so why do we allocate the pPath at all? And for compilers that would reuse the frame space of the pPathStack after it goes out of scope at the end of the block, this is even dangerous.